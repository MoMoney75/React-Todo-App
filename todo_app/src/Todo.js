const Todo = ({task, handleRemove}) =>{
    const remove = ()=>handleRemove()
    return(
        <div>
            {task}
            <button onClick={remove}>x</button>
        </div>
    )
}

export default Todo;
