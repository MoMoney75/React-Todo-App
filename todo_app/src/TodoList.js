import React from "react"
import {useState} from "react" 
import Todo from "./Todo";
import TodoForm from "./NewTodoForm";



const TodoList = function(){

    const [list, setList] = useState([])
    const handleFormSubmit = (formData) =>{
        setList((existingList) =>[...existingList,formData])
    }

    const remove = (key) =>{
        setList(list.filter((task,i) => i !== key ))

    }

    return(
        <div>
            <TodoForm onFormSubmit={handleFormSubmit}/>
        <ol>
        {list.map((task,index) =>(
                
                <li key={index} >
                <Todo 
                task={task.todo}
                handleRemove={()=>remove(index)}
                />
            </li>
        ))}


        </ol>

        </div>
    )
}

export default TodoList;