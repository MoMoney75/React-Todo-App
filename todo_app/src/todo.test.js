import Todo from "./Todo";
import TodoList from "./TodoList";
import {render} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

test("Render Todo component", function(){
    render(<Todo />)
})

test("Snapshot of Todo", function(){
    const {asFragment} = render(<Todo />)
    expect(asFragment()).toMatchSnapshot()
})


test("Render TodoList component", function(){
    render(<TodoList />)
})

test("Snapshot of TodoList", function(){
    const {asFragment} = render(<TodoList />)
    expect(asFragment()).toMatchSnapshot()
})