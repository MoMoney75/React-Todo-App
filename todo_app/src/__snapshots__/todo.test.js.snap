// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of Todo 1`] = `
<DocumentFragment>
  <div>
    <button>
      x
    </button>
  </div>
</DocumentFragment>
`;

exports[`Snapshot of TodoList 1`] = `
<DocumentFragment>
  <div>
    <div>
      <form>
        <label
          for="todo"
        >
          todo:
        </label>
        <input
          id="todo"
          name="todo"
          placeholder="enter task"
          type="text"
          value=""
        />
        <button>
          submit
        </button>
      </form>
    </div>
    <ol />
  </div>
</DocumentFragment>
`;
